amazon-kinesis-client/src/test/java/software/amazon/kinesis/lifecycle/ShardConsumerSubscriberTest.java

```diff
@@ -399,6 +400,7 @@ public void restartAfterRequestTimerExpiresWhenInitialTaskExecutionIsRejected()
         // First try to start subscriptions.
         synchronized (processedNotifier) {
             subscriber.startSubscriptions();
+            processedNotifier.wait(100);
```

Comment:
This code block is duplicated across the tests in this file can we create a method for it with configurable timeout?

Reply:
Correct. In actual application, `ShardConsumer.healthCheck()` would be called within the process loop to retry failed subscriptions. Tests do not use the same retry mechanism, so a wait is required to ensure `healthCheck` successfully performs a restart.